### PR TITLE
python3Packages.pytest: 6.1.2 -> 6.2.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 6.2.2

Change log: https://github.com/pytest-dev/pytest/releases/tag/6.2.2

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

File: pkgs/development/python-modules/pytest/default.nix [hunk 1]
```diff
@@ -75,13 +75,12 @@ buildPythonPackage rec {
     preDistPhases+=" pytestcachePhase"
   '';
 
-  pythonImportsCheck = [
-    "pytest"
-  ];
+  pythonImportsCheck = [ "pytest" ];
```
Comment (FRidh) [inline pkgs/development/python-modules/pytest/default.nix:78]: Please undo this change.
Having items on a separate line is preferable as it makes merge conflicts easier to resolve.

Comment (fabaff) [reply pkgs/development/python-modules/pytest/default.nix]: Expanding the list of import checks of doesn't make much sense as they will all fail if `pytest` can't be imported.

The "one item per line" doesn't correspond with other parts of the file (e.g. the `checkInputs` or the imports).

Reverted and implemented suggestions from `nixpkgs-hammering`.

**Review by davidak — APPROVED** [review]
Change LGTM

I don't have enough disk space to test :grimacing: 
```
[0/53329 built, 16/0/24479 copied (23.0/150767.6 MiB), 28.7/78198.2 MiB DL]
error: interrupted by the user
```

**Comment by SuperSandro2000** [comment]
> Change LGTM
> 
> I don't have enough disk space to test 😬
> 
> ```
> [0/53329 built, 16/0/24479 copied (23.0/150767.6 MiB), 28.7/78198.2 MiB DL]
> error: interrupted by the user
> ```

We usually don't rebuild all packages in a mass rebuild (5000+ rebuilds). You can cherry pick packages like home-assistant or any other more complicated package and only build it with ``-p``.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 113382` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>home-assistant</li>
  </ul>
</details>

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 113382` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python38Packages.pytest-subtests</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Tests are passing with pytest-6.1.2

```bash
============================= test session starts ==============================
platform linux -- Python 3.8.7, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /build/pytest-subtests-0.4.0
plugins: subtests-0.4.0
collected 21 items                                                             

tests/test_subtests.py .s.s.s..s..s.xx......                             [100%]

=================== 14 passed, 5 skipped, 2 xfailed in 0.87s ===================
```

https://github.com/NixOS/nixpkgs/pull/114324 contains an update to 0.4.0 of `pytest-subtests`. I doubt that it helps as upstream claims that they have fixed the `pytest-6` support in 0.3.2.

**Comment by davidak** [comment]
Let's see. Can you rebase?

**Comment by fabaff** [comment]
> Let's see. Can you rebase?

Will do.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113382` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python3Packages.hypothesis</li>
    <li>python3Packages.pytest</li>
    <li>python3Packages.pytest-subtests</li>
  </ul>
</details>

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 113382` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python3Packages.pytest-subtests</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]

**Comment by FRidh** [comment]
In the future, please do not merge any pytest major or minor updates without exhaustive testing of the package set. Updates of this package always result in some breakage, and in this case it resulted in aiohttp failing. Updating Python packages (like these) is difficult because of the regressions they introduce, and is why these are typically done together.

**Comment by davidak** [comment]
@FRidh OK.

How is this testing done in practice? Do you really download 150 GB of packages and build 50000 packages?

**Comment by FRidh** [comment]
We have the `python-unstable` branch that we use about once a month to integrate updates to all Python packages. Many hours of work...

**Comment by SuperSandro2000** [comment]
> How is this testing done in practice? Do you really download 150 GB of packages and build 50000 packages?

I normally just build home-assistant to check if something major is broken.
